src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -817,3 +817,18 @@ def validate_os_sku(namespace):
             CONST_OS_SKU_CBLMARINER,
             CONST_OS_SKU_MARINER,
         )
+
+
+def validate_azure_service_mesh_pluginca(namespace):
+    """ Validates plugin CA parameters for azure service mesh. """
+    if namespace.key_vault_id is not None or namespace.ca_cert_object_name is not None or namespace.ca_key_object_name is not None or namespace.root_cert_object_name is not None or namespace.cert_chain_object_name is not None:
```

Comment:
I'm not sure if the validation is done after **all** command options are parsed, or it's done once **any** command option is parsed. If it is the latter, there may be some problems with this validation. So I would suggest you do the validation **later** in the decorator.

Besides, may take `any([a,b,c,d,e])` to do the check, if you think empty string takes the same effect as None (as Python would implicitly covert "" to false).

Reply:
thanks for the catch, updated as per review comments.